src/abstractnamedgraph.jl

```diff
@@ -381,44 +381,42 @@ function union(graph1::AbstractNamedGraph, graph2::AbstractNamedGraph)
   return union_graph
 end
 
-function add_vertex!(graph::AbstractNamedGraph, vertex)
-  if vertex ∈ vertices(graph)
-    throw(ArgumentError("Duplicate vertices are not allowed"))
-  end
-  add_vertex!(parent_graph(graph))
-  # Update the vertex list
-  push!(vertices(graph), vertex)
-  # Update the reverse map
-  # TODO: Make this more generic
-  insert!(graph.vertex_to_parent_vertex, vertex, last(parent_vertices(graph)))
-  return graph
-end
-
 function rem_vertex!(graph::AbstractNamedGraph, vertex)
-  parent_vertex = vertex_to_parent_vertex(graph, vertex)
-  rem_vertex!(parent_graph(graph), parent_vertex)
-
-  # Insert the last vertex into the position of the vertex
-  # that is being deleted, then remove the last vertex.
-  last_vertex = last(vertices(graph))
-  vertices(graph)[parent_vertex] = last_vertex
-  last_vertex = pop!(vertices(graph))
-
-  # Insert the last vertex into the position of the vertex
-  # that is being deleted, then remove the last vertex.
-  # TODO: Make this more generic
-  graph.vertex_to_parent_vertex[last_vertex] = parent_vertex
-  # TODO: Make this more generic
-  delete!(graph.vertex_to_parent_vertex, vertex)
-
-  return graph
+  if vertex ∉ vertices(graph)
+    return false
+  else
+    parent_vertex = vertex_to_parent_vertex(graph, vertex)
+    rem_vertex!(parent_graph(graph), parent_vertex)
+
+    # Insert the last vertex into the position of the vertex
+    # that is being deleted, then remove the last vertex.
+    last_vertex = last(vertices(graph))
+    vertices(graph)[parent_vertex] = last_vertex
+    last_vertex = pop!(vertices(graph))
+
+    # Insert the last vertex into the position of the vertex
+    # that is being deleted, then remove the last vertex.
+    # TODO: Make this more generic
+    graph.vertex_to_parent_vertex[last_vertex] = parent_vertex
+    # TODO: Make this more generic
+    delete!(graph.vertex_to_parent_vertex, vertex)
+
+    return true
+  end
```

Comment:
```suggestion
  end
    parent_vertex = vertex_to_parent_vertex(graph, vertex)
    rem_vertex!(parent_graph(graph), parent_vertex)

    # Insert the last vertex into the position of the vertex
    # that is being deleted, then remove the last vertex.
    last_vertex = last(vertices(graph))
    vertices(graph)[parent_vertex] = last_vertex
    last_vertex = pop!(vertices(graph))

    # Insert the last vertex into the position of the vertex
    # that is being deleted, then remove the last vertex.
    # TODO: Make this more generic
    graph.vertex_to_parent_vertex[last_vertex] = parent_vertex
    # TODO: Make this more generic
    delete!(graph.vertex_to_parent_vertex, vertex)

    return true
```
Same kind of change elsewhere, like `add_vertex!`. This helps clarify the code logic, to separate out the code into "escape hatches" and then the actual implementation. Also I think it is easier to read code that isn't formatted into an if-statement.